src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,44 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents the Deadline task in the Duke program.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```

Comment:
While this isn't technically a coding standard violation (to my knowledge), it might be clearer to name this variable with a noun, e.g. `date`.

Reply:
I reserved this suggestion as I see this variable as the argument `by` of a `Deadline` task instead of the preposition 'by'. However, you do have a point as it does not sound like a noun.